python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -402,10 +401,38 @@ def configure_at_launch(cls: "Type[Self]", **kwargs) -> "PartialResource[Self]":
         """
         return PartialResource(cls, data=kwargs)
 
-    def initialize_and_run(self, context: InitResourceContext) -> TResValue:
-        # If we have any partially configured resources, we need to update them
-        # with the fully configured resources from the context
+    def _with_updated_values(self, values: Mapping[str, Any]) -> "ConfigurableResource[TResValue]":
+        """
+        Returns a new instance of the resource with the given values.
+        Used when initializing a resource at runtime.
+        """
+        return self.__class__(
+            **{**{k: v for k, v in self.__dict__.items() if not k.startswith("_")}, **values}
```

Comment:
now I feel especially strongly wrapping this since `startswith` appears twice in this PR